x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/authn/SamlAuthnRequestValidator.java

```diff
@@ -0,0 +1,298 @@
+                logAndRespond(new ParameterizedMessage("SAML message [{}] is not an AuthnRequest", SamlUtils.text(root, 128)), listener);
+                return;
+            }
+            final AuthnRequest authnRequest = SamlUtils.buildXmlObject(root, AuthnRequest.class);
+            final SamlServiceProvider sp = getSpFromIssuer(authnRequest.getIssuer());
+            // If the Service Provider should not sign requests, do not try to handle signatures even if they are added to the request
+            if (sp.shouldSignAuthnRequests()) {
+                if (Strings.hasText(signature)) {
+                    if (Strings.hasText(sigAlg) == false) {
+                        logAndRespond(new ParameterizedMessage("Query string [{}] contains a Signature but SigAlg parameter is missing",
+                            queryString), listener);
+                        return;
+                    }
+                    final X509Credential spSigningCredential = sp.getSigningCredential();
+                    if (spSigningCredential == null) {
+                        logAndRespond(
+                            "Unable to validate signature of authentication request, " +
+                                "Service Provider hasn't registered signing credentials",
+                            listener);
+                        return;
+                    }
+                    if (validateSignature(samlRequest, sigAlg, signature, sp.getSigningCredential(), relayState) == false) {
+                        logAndRespond(
+                            new ParameterizedMessage("Unable to validate signature of authentication request [{}] using credentials [{}]",
+                            queryString, SamlUtils.describeCredentials(Collections.singletonList(sp.getSigningCredential()))), listener);
+                        return;
+                    }
+                } else if (Strings.hasText(sigAlg)) {
+                    logAndRespond(new ParameterizedMessage("Query string [{}] contains a SigAlg parameter but Signature is missing",
+                        queryString), listener);
+                    return;
+                } else {
+                    logAndRespond("The Service Provider must sign authentication requests but no signature was found", listener);
+                    return;
+                }
+            }
+            validateAuthnRequest(authnRequest, sp);
+            Map<String, Object> authnState = buildAuthnState(authnRequest, sp);
+            final SamlValidateAuthnRequestResponse response = new SamlValidateAuthnRequestResponse(sp.getEntityId(),
+                authnRequest.isForceAuthn(), authnState);
+            logger.trace(new ParameterizedMessage("Validated AuthnResponse from queryString [{}] and extracted [{}]",
+                queryString, response));
+            listener.onResponse(response);
+        } catch (ElasticsearchSecurityException e) {
+            listener.onFailure(e);
```

Comment:
Should we log something? `validateSignature` throws an exception without logging it (e.g. if there's an invalid signature algorithm), but we don't log it here either.

Reply:
Yes, I handled ESE here explicitly so that we don't wrap it in a new ESE in logAndRespond but forgot to log :/